### PR TITLE
Dvsim tidyups

### PR DESCRIPTION
**Pull request by rswarbrick**
This patch series has two "tidyup" patches (one automatic; one manual) and one actual bug-fix. The latter is the only one that should change behaviour, and it should be pretty uncontroversial! :-)

### CODE REVIEW

File: util/dvsim/dvsim.py [hunk 1]
```diff
@@ -84,14 +83,17 @@ def resolve_branch(arg_branch):
...
-    if proj_root == "":
+    if not proj_root:
         log.error(
-            "Attempted to find the root of this GitHub repository by running:" \
-            "\n..\nBut this command has failed:\n%s" %
-                result.stderr.decode("utf-8"))
+            "Attempted to find the root of this GitHub repository by running:\n"
+            "{}\n"
+            "But this command has failed:\n"
+            "{}".format(' '.join(cmd), result.stderr.decode("utf-8")))
```
Comment (udinator) [inline util/dvsim/dvsim.py]: Minor nit: on lines 54-55 above, you use the `%` syntax to format the string argument of `log.warning()`, and here you use the `.format()` syntax, could you use either one or the other?

Comment (rswarbrick) [reply util/dvsim/dvsim.py]: Ok, I'll change both to use `format`.

File: util/dvsim/dvsim.py [hunk 2]
```diff
@@ -36,8 +35,8 @@
 # Try to create the directory if it does not already exist.
 def resolve_scratch_root(arg_scratch_root):
     scratch_root = os.environ.get('SCRATCH_ROOT')
-    if arg_scratch_root == None or arg_scratch_root == "":
-        if scratch_root == None:
+    if not arg_scratch_root:
+        if scratch_root is None:
```
Comment (sriyerg) [inline util/dvsim/dvsim.py]: Is there a guidance on when to use `is` and when to use `==`?

Comment (rswarbrick) [reply util/dvsim/dvsim.py]: Yes. https://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is

Comment (rswarbrick) [reply util/dvsim/dvsim.py]: For a more canonical answer, the [PEP8](https://www.python.org/dev/peps/pep-0008) text is pretty readable. In particular, see the "Programming Recommendations" section.

**Pull request merged**